src/maxi/utils/loss_utils.py

```diff
@@ -184,19 +184,13 @@ def torch_extract_target_proba(P: torch.Tensor, t: int) -> torch.Tensor:
     Returns:
         tf.Tensor: Prediction score for target class
     """
-    if P.ndim not in [1, 2]:
-        raise ValueError("Got logits matrix of dimension larger than 2")
-
-    if P.ndim == 1:
-        return P[t]
+    if P.ndim != 2:
+        raise ValueError(f"Prediction should have dimension 2, but got {P.ndim}")
 
-    target_vector = torch.zeros(len(P), dtype=torch.float32)
-    for i in range(len(P)):
-        target_vector[i] = torch_extract_target_proba(P[i], t)
-    return target_vector
+    return P[:, t]
 
 
-def torch_extract_nontarget_proba(P: torch.Tensor, t: int) -> torch.Tensor:
+def torch_extract_nontarget_proba(P: torch.Tensor, t: torch.int64) -> torch.Tensor:
```

Comment:
Does this also work when t is 0 or -1?

Reply:
For 0 it does, for -1 most probably not. Valid values should be: [0, t.shape[1]-1]. If we keep it this way, we might have two options:

- add a safety check to be sure we are not out of bound for the value of t
- add tests to be sure the behavior is as expected (generally advisable)

Reply:
Solved with commit 62c4f17